### PR TITLE
Port concordance to 0.2

### PR DESCRIPTION
**Pull request by jigold**
@lfrancioli -- here's the 0.2 version of concordance

I wasn't sure how to handle describing the requirements for the entry schema. Will it assume there is an entry field "GT" with type Call? For now, I left it the same documentation as `dataset.py`.

### CODE REVIEW

**Comment by konradjk** [comment]
I would have assumed you'd indicate what to do concordance on (e.g. parameters to be `left.GT` and `right.GT` though I haven't thought about if the join will be tricky once you allow arbitrary row and col keys)

File: python/hail/methods/tests.py [hunk 1]
```diff
@@ -100,3 +100,12 @@ def test_mendel_errors(self):
...
         dataset.annotate_rows(mendel=var[dataset.v]).count_rows()
+
+    def test_concordance(self):
+        dataset = self.get_dataset()
+        glob_conc, cols_conc, rows_conc = methods.concordance(dataset, dataset)
+        print(glob_conc[1][4])
+        print(glob_conc[4][0])
+        print(glob_conc[:][3])
+        cols_conc.write('/tmp/foo.kt', overwrite=True)
+        rows_conc.write('/tmp/foo.kt', overwrite=True)
```
Comment (patrick-schultz) [inline python/hail/methods/tests.py]: I assume this wasn't supposed to stay in?

Comment (jigold) [reply python/hail/methods/tests.py]: Why? Table has a `write` method.

Comment (patrick-schultz) [reply python/hail/methods/tests.py]: Sorry, that window is misleading, I meant the entire test. There are print statements, and no assertions.

**Comment by konradjk** [comment]
What was the conclusion on this?:

> Will it assume there is an entry field "GT" with type Call?

**Pull request merged**